### PR TITLE
Local Income Expense - New options

### PR DESCRIPTION
**Pull request by cmboyman**
https://github.com/digi-serve/ns_app/issues/234

### CODE REVIEW

File: reports/local-income-expense.js [hunk 1]
```diff
@@ -3,21 +3,27 @@
  *
  *
  */
+const { reject } = require("async");
```
Comment (wongpratan) [inline reports/local-income-expense.js:6]: Do we need it for what line ?

File: reports/local-income-expense.js [hunk 2]
```diff
@@ -258,26 +353,69 @@ module.exports = {
...
+            rule: "contains",
+            value: fyYear,
+         });
+      }
+
+      if (fyMonth) {
+         let CurrentYear = new Date().getFullYear();
+
+         if (fyYear === undefined) {
+            fyYear = CurrentYear;
```
Comment (wongpratan) [inline reports/local-income-expense.js:402]: Should move `let CurrentYear = new Date().getFullYear();` to here ?

File: reports/local-income-expense.js [hunk 3]
```diff
@@ -218,16 +312,17 @@ module.exports = {
             });
 
             if (!rc) {
-               rc = data.rcOptions[0];
+               rc = data.rcOptions[-1];
```
Comment (wongpratan) [inline reports/local-income-expense.js:315]: Is it always `null` from `[-1]` ?

File: reports/templates/local-income-expense.ejs [hunk 4]
```diff
@@ -1,26 +1,55 @@
 <%- include('/app/reports/templates/stylesheets') %>
+<script type="text/javascript">
+      function getMonthVal(){
+         let monthVal = document.getElementById("monthSelect").value;
+         let yearVal = document.getElementById("yearSelect").value;
+         location.href = '/report/local-income-expense?fyYear='+yearVal+'&fyMonth='+monthVal;
```
Comment (wongpratan) [inline reports/templates/local-income-expense.ejs:6]: Use string format ?

File: reports/local-income-expense.js [hunk 5]
```diff
@@ -258,26 +353,69 @@ module.exports = {
...
+      }
+
+      if (fyMonth) {
+         let CurrentYear = new Date().getFullYear();
+
+         if (fyYear === undefined) {
+            fyYear = CurrentYear;
+         }
+
+         const monthJoin = fyYear + " M" + fyMonth;
```
Comment (wongpratan) [inline reports/local-income-expense.js:405]: Should use string format ?

File: reports/local-income-expense.js [hunk 6]
```diff
@@ -258,26 +353,69 @@ module.exports = {
...
+         }
+         where.rules.push({
+            key: "RC Code",
+            rule: "equals",
+            value: teamListObjs,
+         });
+      }
+
+      if (rc) {
+         if (rc === "all") {
```
Comment (wongpratan) [inline reports/local-income-expense.js:380]: `rc !== "all"`

File: reports/local-income-expense.js (hunk 1 above)
Comment (cmboyman) [reply reports/local-income-expense.js:6]: fixed.

File: reports/local-income-expense.js (hunk 2 above)
Comment (cmboyman) [reply reports/local-income-expense.js:402]: fixed.

File: reports/local-income-expense.js (hunk 3 above)
Comment (cmboyman) [reply reports/local-income-expense.js:315]: fixed.

**Comment by wongpratan** [comment]
It might duplicate to #fixlocalIncomeOption

There are update/resolve code in #fixlocalIncomeOption
https://github.com/digi-serve/ab_service_custom_reports/commit/cb03e6c11bd7e91e8c5378a667d34ba985ce618a

We can close this PR, then we will merge #fixlocalIncomeOption instead

**Pull request closed without merging**